app/api/startBuy/route.ts

```diff
@@ -0,0 +1,34 @@
+// @@@SNIPSTART typescript-next-oneclick-api
+import { OneClickBuy } from '../../../temporal/src/workflows';
+import { getTemporalClient } from '../../../temporal/src/client';
+import { TASK_QUEUE_NAME } from '../../../temporal/src/shared';
+
+export async function POST(req: Request) {
+  interface RequestBody {
+    itemId: string;
+    transactionId: string;
+  }
+
+  let body: RequestBody;
+
+  try {
+    body = await req.json() as RequestBody;
+  } catch (error) {
+    return new Response("Invalid JSON body", { status: 400 });
+  }
+
+  const { itemId, transactionId } = body;
+
+  if (!itemId) {
+    return new Response("Must send the itemID to buy", { status: 405 });
```

Comment:
405 is 'method not allowed' that doesn't super match with !itemId. How about we also use 400 insted?

Reply:
Agree.